docs/guide/mocking.md

```diff
@@ -1,293 +1,274 @@
+Add the following to your test [setup file](/config/#setupfiles)
+```js
+import { beforeAll, afterAll, afterEach } from 'vitest'
+import { setupServer } from 'msw/node'
+import { graphql, rest } from 'msw'
 
-```ts
-test('async test', async () => {
-  const asyncMock = vi
-    .fn()
-    .mockResolvedValue('default')
-    .mockResolvedValueOnce('first call')
-    .mockResolvedValueOnce('second call');
+const posts = [
+  {
+    userId: 1,
+    id: 1,
+    title: 'first post title',
+    body: 'first post body',
+  },
+  ...
+]
 
-  await asyncMock(); // first call
-  await asyncMock(); // second call
-  await asyncMock(); // default
-  await asyncMock(); // default
-});
-```
+export const restHandlers = [
+  rest.get('https://rest-endpoint.example/path/to/posts', (req, res, ctx) => {
+    return res(ctx.status(200), ctx.json(posts))
+  }),
+]
 
-### mockRejectedValue
+const graphqlHandlers = [
+  graphql.query('https://graphql-endpoint.example/api/v1/posts', (req, res, ctx) => {
+    return res(ctx.data(posts))
+  }),
+]
 
-**Type:** `(value: any) => MockInstance`
+const server = setupServer(...restHandlers, ...graphqlHandlers)
 
-Accepts an error that will be rejected, when async function will be called.
+// Start server before all tests
+beforeAll(() => server.listen({ onUnhandledRequest: 'error' }))
 
-```ts
-test('async test', async () => {
-  const asyncMock = vi.fn().mockRejectedValue(new Error('Async error'));
+//  Close server after all tests
+afterAll(() => server.close())
 
-  await asyncMock(); // throws "Async error"
-});
+// Reset handlers after each test `important for test isolation`
+afterEach(() => server.resetHandlers())
 ```
 
-### mockRejectedValueOnce
-
-**Type:** `(value: any) => MockInstance`
+> Configuring the server with `onUnhandleRequest: 'error'` ensures that an error is thrown whenever there is a request that does not have a corresponding request handler.
 
-Accepts a value that will be rejected for one call to the mock function. If chained, every consecutive call will reject passed value.
+### Example
 
-```ts
-test('async test', async () => {
-  const asyncMock = vi
-    .fn()
-    .mockResolvedValueOnce('first call')
-    .mockRejectedValueOnce(new Error('Async error'));
+We have a full working example which uses MSW: [React Testing with MSW](https://github.com/vitest-dev/vitest/tree/main/test/react-testing-lib-msw).
 
-  await asyncMock(); // first call
-  await asyncMock(); // throws "Async error"
-});
-```
+### More
+There is much more to MSW. You can access cookies and query parameters, define mock error responses, and much more! To see all you can do with MSW, read [their documentation](https://mswjs.io/docs/recipes).
 
-## Mock properties
+## Timers
 
-### mock.calls
+Whenever we test code that involves timeOuts or intervals, instead of having our tests it wait out or time-out. We can speed up our tests by using 'fake timers'. You can mock calls to `setTimeout` and `setInterval` easily to
```

Comment:
```suggestion
Whenever we test code that involves `timeOut`s or intervals, instead of having our tests it wait out or time-out. We can speed up our tests by using "fake" timers by mocking calls to `setTimeout` and `setInterval`, too.
```